trieste/models/gpflow/builders.py

```diff
@@ -474,3 +475,108 @@ def build_multifidelity_autoregressive_models(
     ]
 
     return gprs
+
+
+def build_multifidelity_nonlinear_autoregressive_models(
+    dataset: Dataset,
+    num_fidelities: int,
+    input_search_space: SearchSpace,
+    kernel_base_class: Type[Stationary] = gpflow.kernels.Matern32,
+) -> Sequence[GaussianProcessRegression]:
+    """
+    Build models for training the trieste.models.gpflow.MultifidelityNonlinearAutoregressive` model
+
+    Builds a basic Matern32 kernel for the lowest fidelity, and the custom kernel described in
+    :cite:`perdikaris2017nonlinear` for the higher fidelities, which also have an extra input
+    dimension. Note that the initial data that the models with fidelity greater than 0 are
+    initialised with contain dummy data in this extra dimension, and so an `update` of the
+    `MultifidelityNonlinearAutoregressive` is required to propogate real data through to these
+    models.
+
+    :param dataset: The dataset to use to initialise the models
+    :param fidelities: The number of fidelities to model
+    :return: gprs: A list containing gprs that can be used for the multifidelity model
+    """
+    # Split data into fidelities
+    data = split_dataset_by_fidelity(dataset=dataset, num_fidelities=num_fidelities)
+
+    _validate_multifidelity_data_modellable(data, num_fidelities)
+
+    # Input dim requires excluding fidelity row
+    input_dim = dataset.query_points.shape[1] - 1
+
+    # Create kernels
+    kernels = _create_multifidelity_nonlinear_autoregressive_kernels(
```

Comment:
looks like you can't change `ard` option to `True`, so need to make it also an optional argument of the builder?